.github/workflows/dependency_modification_check.yml

```diff
@@ -0,0 +1,18 @@
+name: Check no dependencies were modified
+
+on:
+  pull_request
+
+jobs:
+  dependency_changed_check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: "Check Cargo.lock not in changeset" 
```

Comment:
It can occur where the `Cargo.lock` is included in the PR but `Cargo.toml` files are updated such that when built it updates the `Cargo.lock`. This test misses this case. This case is caught with https://github.com/firecracker-microvm/firecracker/pull/3631/commits/28129693a269c1f325665377ca396ff14afddb61 but this would mean that when we run CI if the lockfile is out-of-date it causes fails in many tests which don't need to care, it would be preferable to have only 1 failure in 1 test for this 1 error case (such that if other error cases occurred we could see them and fix them before needing another CI re-run).

This test can instead use `run: cargo generate-lockfile --locked`

Reply:
Ahh, that works around the issue of not having control over test execution order, nice! Is there a way to mark a GitHub Action as required? I think we'll still want to block CI on out-of-date lockfiles, and the dependency_modification_check one is an optional action.

Reply:
Ah, amazing, thanks! I'll drop 28129693a269c1f325665377ca396ff14afddb61 then and add the github action you propose then :) 

putting this in draft until that's done

Reply:
done!